lms/djangoapps/django_comment_client/permissions.py

```diff
@@ -55,10 +56,35 @@ def check_question_author(user, content):
         except KeyError:
             return False
 
+    def check_team_member(user, content):
+        """
+        If the content has a commentable_id, verifies that either it is not associated with a team,
+        or if it is, that the user is a member of that team.
+        """
+        if not content:
+            return False
+        try:
+            commentable_id = content['commentable_id']
```

Comment:
Note that we could avoid this sql query for non-team threads if we extend @benmcmorran's work and store the course vs. standalone context. I'm starting to think that is worth doing.

Another possibility would be to only check for team status if the commentable_id starts with the prefix of team discussion topics, but that feels too much like a hack and is brittle if we change the naming convention in the future.